### PR TITLE
Fix HA config update on an existing cluster

### PR DESCRIPTION
**Pull request by a-palchikov**
## Description

Fixes an issue with the HA configuration update on an existing cluster.
While testing planet changes to accommodate the tip of coordinate (which was updated based on work on the [reconcile loop](https://github.com/gravitational/planet/pull/575) which was never merged) I stumbled on this behavior.

Given this configuration:
```
# config.yaml
kind: ClusterConfiguration
version: v1
spec:
  global:
    cloudProvider: generic
    highAvailability: true
```

Before the change:
```
$ gravity resource create -f ./haconfig.yaml --confirm
[ERROR]: provided cluster configuration is empty
```

After the change:
```
$ gravity resource create -f haconfig.yaml --confirm
gravity resource create -f ./haconfig.yaml --confirm
Mon Jul 19 13:29:51 UTC	Deploying agents on cluster nodes
Deployed agent on vm-2 (172.17.0.3)
Deployed agent on vm-3 (172.17.0.4)
Deployed agent on vm-1 (172.17.0.2)
Executing "/update-config" locally
Mon Jul 19 13:30:02 UTC	Update runtime configuration
Executing "/masters/vm-1/stepdown" locally
Mon Jul 19 13:30:04 UTC	Step down "vm-1" as Kubernetes leader
Disable leader elections on vm-1
Waiting for new leader election
Executing "/masters/vm-1/drain" locally
Mon Jul 19 13:30:06 UTC	Drain node "vm-1"
	Still executing "/masters/vm-1/drain" locally (10 seconds elapsed)
Executing "/masters/vm-1/restart" locally
Mon Jul 19 13:30:18 UTC	Restart container on node "vm-1"
	Still executing "/masters/vm-1/restart" locally (10 seconds elapsed)
	Still executing "/masters/vm-1/restart" locally (20 seconds elapsed)
	Still executing "/masters/vm-1/restart" locally (30 seconds elapsed)
	Still executing "/masters/vm-1/restart" locally (40 seconds elapsed)
	Still executing "/masters/vm-1/restart" locally (50 seconds elapsed)
	Still executing "/masters/vm-1/restart" locally (1 minute elapsed)
	Still executing "/masters/vm-1/restart" locally (1 minute elapsed)
	Still executing "/masters/vm-1/restart" locally (1 minute elapsed)
	Still executing "/masters/vm-1/restart" locally (1 minute elapsed)
	Still executing "/masters/vm-1/restart" locally (1 minute elapsed)
Executing "/masters/vm-1/elect" locally
Mon Jul 19 13:31:58 UTC	Make node "vm-1" Kubernetes leader
Disable leader elections on vm-2
Disable leader elections on vm-3
Enable leader elections on vm-1
Waiting for new leader election
Executing "/masters/vm-1/taint" locally
Mon Jul 19 13:32:00 UTC	Taint node "vm-1"
	Still executing "/masters/vm-1/taint" locally (10 seconds elapsed)
Executing "/masters/vm-1/uncordon" locally
Mon Jul 19 13:32:17 UTC	Uncordon node "vm-1"
Executing "/masters/vm-1/endpoints" locally
Mon Jul 19 13:32:19 UTC	Wait for endpoints on node "vm-1"
Executing "/masters/vm-1/untaint" locally
Mon Jul 19 13:32:20 UTC	Remove taint from node "vm-1"
Executing "/masters/vm-2/drain" on remote node vm-2
Mon Jul 19 13:32:21 UTC	Drain node "vm-2"
Executing "/masters/vm-2/restart" on remote node vm-2
Mon Jul 19 13:32:29 UTC	Restart container on node "vm-2"
	Still executing "/masters/vm-2/restart" on remote node vm-2 (10 seconds elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (20 seconds elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (30 seconds elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (40 seconds elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (50 seconds elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (1 minute elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (1 minute elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (1 minute elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (1 minute elapsed)
	Still executing "/masters/vm-2/restart" on remote node vm-2 (1 minute elapsed)
Executing "/masters/vm-2/taint" on remote node vm-2
Mon Jul 19 13:34:11 UTC	Taint node "vm-2"
Executing "/masters/vm-2/uncordon" on remote node vm-2
Mon Jul 19 13:34:13 UTC	Uncordon node "vm-2"
Executing "/masters/vm-2/endpoints" on remote node vm-2
Mon Jul 19 13:34:16 UTC	Wait for endpoints on node "vm-2"
Executing "/masters/vm-2/untaint" on remote node vm-2
Mon Jul 19 13:34:19 UTC	Remove taint from node "vm-2"
Executing "/masters/vm-2/enable-elections" on remote node vm-2
Mon Jul 19 13:34:22 UTC	Enable leader election on node "vm-2"
Executing "/masters/vm-3/drain" on remote node vm-3
Mon Jul 19 13:34:25 UTC	Drain node "vm-3"
Executing "/masters/vm-3/restart" on remote node vm-3
Mon Jul 19 13:34:31 UTC	Restart container on node "vm-3"
	Still executing "/masters/vm-3/restart" on remote node vm-3 (10 seconds elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (20 seconds elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (30 seconds elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (40 seconds elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (50 seconds elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (1 minute elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (1 minute elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (1 minute elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (1 minute elapsed)
	Still executing "/masters/vm-3/restart" on remote node vm-3 (1 minute elapsed)
Executing "/masters/vm-3/taint" on remote node vm-3
Mon Jul 19 13:36:13 UTC	Taint node "vm-3"
Executing "/masters/vm-3/uncordon" on remote node vm-3
Mon Jul 19 13:36:15 UTC	Uncordon node "vm-3"
Executing "/masters/vm-3/endpoints" on remote node vm-3
Mon Jul 19 13:36:17 UTC	Wait for endpoints on node "vm-3"
Executing "/masters/vm-3/untaint" on remote node vm-3
Mon Jul 19 13:36:20 UTC	Remove taint from node "vm-3"
Executing "/masters/vm-3/enable-elections" on remote node vm-3
Mon Jul 19 13:36:23 UTC	Enable leader election on node "vm-3"
operation(update configuration(4cc884e6-553c-44dc-af28-01920e1aae8a), cluster=test, created=2021-07-19 13:29) finished in 6 minutes
```

## Type of change

* Bug fix (non-breaking change which fixes an issue)
* This change has a user-facing impact

## Linked tickets and other PRs

* Requires https://github.com/gravitational/planet/pull/855, https://github.com/gravitational/coordinate/pull/15

## TODOs

- [x] Self-review the change
- [x] Perform manual testing
- [ ] Address review feedback
- [ ] Update upstream references / tags / versions after upstream PR merges (linked above)

## Testing done

Tested with multiple HA configuration updates on a 3-node cluster (all control plane nodes).

### CODE REVIEW

File: versions.mk [hunk 1]
```diff
@@ -10,7 +10,8 @@ K8S_VER ?= 1.21.2
 # major + minor padded to 2 chars with 0 + patch also padded to 2 chars, e.g.
 # 1.13.5 -> 11305, 1.13.12 -> 11312, 2.0.0 -> 20000 and so on
 K8S_VER_SUFFIX ?= $(shell printf "%d%02d%02d" $(shell echo $(K8S_VER) | sed "s/\./ /g"))
-PLANET_TAG ?= 9.0.3-$(K8S_VER_SUFFIX)
+#PLANET_TAG ?= 9.0.4-$(K8S_VER_SUFFIX)
+PLANET_TAG ?= 9.0.3-$(K8S_VER_SUFFIX)-3-g87470a2
```
Comment (wadells) [inline versions.mk:14]: Is this related to the rest of the changes?  It isn't clear to me how.

**Review by wadells — APPROVED** [review]

File: versions.mk (hunk 1 above)
Comment (a-palchikov) [reply versions.mk:14]: This includes the actual coordinate package bump.

**Review by bernardjkim — APPROVED** [review]
